src/overlays/actors/ovl_Bg_Astr_Bombwall/z_bg_astr_bombwall.c

```diff
@@ -28,59 +36,194 @@ const ActorInit Bg_Astr_Bombwall_InitVars = {
-    2, D_80C0A620, // sTrisElementsInit,
+static ColliderTrisInit sTrisInit = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_NONE,
+        OC2_NONE,
+        COLSHAPE_TRIS,
+    },
+    2,
+    sTrisElementsInit,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80C0A6A8[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderTrisElementInit D_80C0A620[2];
-extern ColliderTrisInit D_80C0A698;
-extern InitChainEntry D_80C0A6A8[];
-
-extern UNK_TYPE D_06002178;
-extern UNK_TYPE D_06002380;
-extern UNK_TYPE D_06002498;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C09ED0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/BgAstrBombwall_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/BgAstrBombwall_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A120.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A378.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A38C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A400.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A418.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A458.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/func_80C0A4BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/BgAstrBombwall_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Astr_Bombwall/BgAstrBombwall_Draw.s")
+extern Gfx D_06002178;
+extern CollisionHeader D_06002498;
+extern Gfx D_06002380[];
+extern Gfx D_060022E0[];
+
+void func_80C09ED0(ColliderTrisInit* arg0, Vec3f* arg1, Vec3s* arg2, ColliderTris* arg3) {
+    s32 i;
+    s32 j;
+    Vec3f sp54[3];
+
+    Matrix_RotateYS(arg2->y, MTXMODE_NEW);
+    Matrix_RotateXS(arg2->x, MTXMODE_APPLY);
+    Matrix_RotateZS(arg2->z, MTXMODE_APPLY);
+
+    for (i = 0; i < arg0->count; i++) {
+        for (j = 0; j < 3; j++) {                                          // https://decomp.me/scratch/JrEnl
+            Matrix_MultVec3f(&(arg0->elements + i)->dim.vtx[j], &sp54[j]); //! FAKE MATCH:
+            Math_Vec3f_Sum(&sp54[j], arg1, &sp54[j]);
+        }
+        Collider_SetTrisVertices(arg3, i, &sp54[0], &sp54[1], &sp54[2]);
+    }
+}
+
+void BgAstrBombwall_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderTris* temp_a1;
+    BgAstrBombwall* this = THIS;
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init((DynaPolyActor*)&this->dyna.actor, 1);
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor*)&this->dyna.actor, &D_06002498);
+    temp_a1 = &this->unk160;
+    Collider_InitTris(globalCtx, temp_a1);
```

Comment:
```suggestion
    Collider_InitTris(globalCtx, &this->unk160);
```
Is this temp necessary, or can it be replaced? (I think this is the collider we typically name this `collider`)